### PR TITLE
Upgrade to okta-auth-js 5.0

### PR DESCRIPTION
**Pull request by denysoblohin-okta**
Issue: [OKTA-386969](https://oktainc.atlassian.net/browse/OKTA-386969)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# 3.3.0
```
Comment (shuowu) [inline CHANGELOG.md:1]: should be 3.2.0

**Review by shuowu — CHANGES_REQUESTED** [review]
~It would be safe to also make this one as a breaking change, as okta-angular exposes [observable authstate](https://github.com/okta/okta-angular#oktaauthauthenticationstate) which devs code may depend on.
It maybe won't be `null` as what okta-react behaves (check needed), but the `isPending` status has been removed for sure.~

Edit: sorry about my overlook... okta-angular exposes the `$authenticationState` as an observable boolean, instead of an object. Then this change should be a minor version update. Basically maintainance change to upgrade to auth-js 5.0.

File: package.json [hunk 2]
```diff
@@ -60,7 +60,7 @@
     "@rollup/plugin-node-resolve": "7.0.0",
     "@types/jest": "^26.0.14",
     "@types/json-schema": "^7.0.3",
-    "@types/node": "^14.11.2",
+    "@types/node": "10.14.0",
```
Comment (shuowu) [inline package.json:63]: looks like a downgrade, what happens here?

File: yarn-error.log [hunk 3]
```diff
@@ -1,12576 +0,0 @@
-Arguments:
```
Comment (shuowu) [inline yarn-error.log:1]: can we add this one to .gitignore?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,9 @@
+# 3.2.0
+
+### Features
```
Comment (shuowu) [inline CHANGELOG.md:3]: Let's use `### Other` for dependency upgrade

File: src/okta/services/okta.service.ts [hunk 5]
```diff
@@ -91,14 +93,26 @@ export class OktaAuthService extends OktaAuth {
...
 
+    ngOnDestroy(): void {
+      this.stop();
+    }
+
     private async emitAuthenticationState(state: boolean) {
       this.observers.forEach(observer => observer.next(state));
     }
 
+    public async isAuthenticated(): Promise<boolean> {
```
Comment (shuowu) [inline src/okta/services/okta.service.ts:109]: 👍 
before this change the config option is not used in code?

**Review by shuowu — APPROVED** [review]

File: src/okta/services/okta.service.ts (hunk 5 above)
Comment (shuowu) [reply src/okta/services/okta.service.ts:109]: let's also add a test for it

File: package.json (hunk 2 above)
Comment (shuowu) [reply package.json:63]: looks like SIW need to explicitly install @types/node, https://github.com/okta/okta-developer-docs/pull/2284/files#diff-25fddbf1b579b42f71891a504022c8ea3c09ad6b9697ecc38e2e47de3fea43d6R63

can you explain more about that? maybe we want it be in dependencies?

**Comment by mraible** [comment]
Ship it! 😃

File: src/okta/services/okta.service.ts (hunk 5 above)
Comment (denysoblohin-okta) [reply src/okta/services/okta.service.ts:107]: There is already test for this: https://github.com/okta/okta-angular/blob/master/test/spec/service.test.ts#L150
It was failing after upgrade to okta-auth-js 5.
Reason - the implementation of `OktaAuth.isAuthenticated()` has been changed:
https://github.com/okta/okta-auth-js/blob/4.8/lib/browser/browser.ts#L443
https://github.com/okta/okta-auth-js/blob/5.0/lib/OktaAuth.ts#L484

**Review by shuowu — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (shuowu) [reply package.json:63]: it's fine if we don't depend on it in bundles.

File: package.json [hunk 6]
```diff
@@ -42,7 +42,7 @@
   ],
   "license": "Apache-2.0",
   "dependencies": {
-    "@okta/okta-auth-js": "^4.8.0",
+    "@okta/okta-auth-js": "^5.0.0",
```
Comment (aarongranick-okta) [inline package.json:45]: Please use latest version 5.2.0

**Review by aarongranick-okta — APPROVED** [review]

**Pull request closed without merging**